src/superphot_plus/samplers/numpyro_sampler.py

```diff
@@ -0,0 +1,328 @@
+        A truncated normal distribution.
+    """
+    return dist.TruncatedNormal(
+        loc=fields.mean, scale=fields.std, low=fields.clip_a, high=fields.clip_b, validate_args=True
+    )
+
+
+def run_mcmc(lc, sampler="NUTS", priors=MultibandPriors.load_ztf_priors()):
+    """Runs MCMC using numpyro on the lightcurve to get set
+    of equally weighted posteriors (sets of fit parameters).
+
+    Parameters
+    ----------
+    lc : Lightcurve object
+        The Lightcurve object on which to run MCMC
+    sampler : str, optional
+        The MCMC sampler to use. Defaults to "NUTS".
+    priors : MultibandPriors, optional
+        The prior set to use for fitting. Defaults to ZTF's priors.
+
+    Returns
+    -------
+    np.ndarray or None
+        A set of equally weighted posteriors (sets of fit parameters) as
+        a numpy array. If the lightcurve does not contain any valid
+        points, None is returned.
+    """
+    # Require data in all bands.
+    for unique_band in priors.ordered_bands:
+        if lc.obs_count(unique_band) == 0:
+            return None
+
+    def jax_model(t=None, obsflux=None, uncertainties=None, max_flux=None):
+        """JAX model for MCMC.
+
+        Parameters
+        ----------
+        t : array-like, optional
+            Time values. Defaults to None.
+        obsflux : array-like, optional
+            Observed flux values. Defaults to None.
+        uncertainties : array-like, optional
+            Flux uncertainties. Defaults to None.
+        max_flux : float, optional
+            Maximum flux value. Defaults to None.
+        """
+        ref_priors = priors.bands[priors.reference_band]
+
+        A, beta, gamma, t0, tau_rise, tau_fall, extra_sigma = prior_helper(ref_priors, max_flux)
+
+        phase = t - t0
+        flux_const = A / (1.0 + jnp.exp(-phase / tau_rise))
+        sigmoid = 1 / (1 + jnp.exp(10.0 * (gamma - phase)))
+
+        flux = flux_const * (
+            (1 - sigmoid) * (1 - beta * phase)
+            + sigmoid * (1 - beta * gamma) * jnp.exp(-(phase - gamma) / tau_fall)
+        )
+        sigma_tot = jnp.sqrt(uncertainties**2 + extra_sigma**2)
+
+        # auxiliary bands
+        for b_idx, uniq_b in enumerate(priors.aux_bands):
+            b_priors = priors.bands[uniq_b]
+
+            (
+                _,
+                beta_ratio,
+                gamma_ratio,
+                t0_ratio,
+                tau_rise_ratio,
+                tau_fall_ratio,
+                extra_sigma_ratio,
+            ) = prior_helper(b_priors, max_flux, uniq_b)
+
+            beta_b = beta * beta_ratio
+            gamma_b = gamma * gamma_ratio
+            t0_b = t0 * t0_ratio
+            tau_rise_b = tau_rise * tau_rise_ratio
+            tau_fall_b = tau_fall * tau_fall_ratio
+
+            inc_band_ix = np.arange(b_idx * PAD_SIZE, (b_idx + 1) * PAD_SIZE)
+
+            phase_b = (t - t0_b)[inc_band_ix]
+            flux_const_b = A / (1.0 + jnp.exp(-phase_b / tau_rise_b))
```

Comment:
Oh! This is a bug, should be A_b, and then line 159 should be A_ratio, with A_b = A * A_ratio